### DISCUSSION TITLE
add listener to streamProvider inside initState

### DISCUSSION THREAD

**Discussion by vrman** [rrousselGit/riverpod#2633] [Q&A]
i used this code to add listener for streamProvider inside any functions like initState or ... 
```
ref.read(locationStreamProvider.stream).listen((value) {  });
```
and it works ok , but recently i go this problem message :

> 'stream' is deprecated and shouldn't be used. .stream will be removed in 3.0.0. As a replacement, either listen to the provider itself (AsyncValue) or .future.

what is alternative way for that ?

**Comment by rrousselGit** [comment]
Use ref.listen instead. Providers have built in listening mechanism

**Reply by vrman** [reply]
but i want to do that inside a function , i got this message :

> ref.listen can only be used within the build method of a ConsumerWidget

**Reply by Trung15010802** [reply]
Try ref.listenManual()
